packages/cdktf-cli/bin/cmds/terraform-check.ts

```diff
@@ -1,9 +1,21 @@
 import { Terraform } from './ui/models/terraform'
+import * as semver from 'semver';
+
+const MIN_SUPPORTED_VERSION = '0.12.0'
 
 export const terraformCheck = async (): Promise<void> => {
   try {
-    const terraform = new Terraform('./')
-    await terraform.version()
+    const terraform: Terraform = new Terraform('./')
+
+    // string in format of 'Terraform v0.13.3'
+    const terraformVersion = await terraform.version()
+    const cleanTerraformVersion = semver.clean(terraformVersion.substring(terraformVersion.indexOf('v')))
```

Comment:
I've tried `0.13.0`, `0.12.29` and `0.11.14`, where I'd have expected the last one to fail. However, it didn't. It looks like `cleanTerraformVersion` is always null.

Some logging for this `terraformVersion`:

```
{
  terraformVersion: 'Terraform v0.11.14\n' +
    '+ provider.aws v2.70.0\n' +
    '\n' +
    'Your version of Terraform is out of date! The latest version\n' +
    'is 0.13.4. You can update by downloading from www.terraform.io/downloads.html\n',
}
```

Looks like `semver.clean` doesn't handle arbritary strings. Perhaps it would make sense to parse `Terraform v<version regex>` explicitly und use the match group as input for the semver parsing. What do you think?

Reply:
@skorfmann sorry have been swamped at work the previous days. Yes that makes sense since there might be other ways of formatting the version message. I'll check to implement the change.